maat-court-data-api/src/main/java/gov/uk/courtdata/laastatus/builder/CourtDataDTOBuilder.java

```diff
@@ -55,7 +55,7 @@ public CourtDataDTO build(CaseDetails caseDetails) {
             offence.setOffenceId(offenceEntity.map(OffenceEntity::getOffenceId).orElse(null));
 
             log.info("legal status {}, date {}", offence.getLegalAidStatus(), offence.getLegalAidStatusDate());
-            if (!offence.getLegalAidStatus().equals("AP") && offence.getLegalAidStatusDate() == null) {
+            if (!offence.getLegalAidStatus().equals("AP") && offence.getLegalAidStatusDate() != null) {
```

Comment:
I think we also need to switch the order of these statements so the `!= null` is tested first.